tokens/src/commands.rs

```diff
@@ -492,61 +499,79 @@ fn read_allocations(
     transfer_amount: Option<u64>,
     require_lockup_heading: bool,
     raw_amount: bool,
-) -> io::Result<Vec<Allocation>> {
+) -> Result<Vec<Allocation>, Error> {
     let mut rdr = ReaderBuilder::new().trim(Trim::All).from_path(input_csv)?;
     let allocations = if let Some(amount) = transfer_amount {
-        let recipients: Vec<String> = rdr
-            .deserialize()
-            .map(|recipient| recipient.unwrap())
-            .collect();
-        recipients
-            .into_iter()
-            .map(|recipient| Allocation {
-                recipient,
-                amount,
-                lockup_date: "".to_string(),
+        rdr.deserialize()
+            .map(|recipient| {
+                let recipient: String = recipient?;
+                match Pubkey::from_str(recipient.as_str()) {
+                    Ok(..) => (),
+                    Err(err) => return Err(Error::BadInputPubkeyError(err)),
+                }
+                Ok(Allocation {
+                    recipient,
+                    amount,
+                    lockup_date: "".to_string(),
+                })
             })
-            .collect()
+            .collect::<Result<Vec<Allocation>, Error>>()?
     } else if require_lockup_heading {
-        let recipients: Vec<(String, f64, String)> = rdr
-            .deserialize()
-            .map(|recipient| recipient.unwrap())
-            .collect();
-        recipients
-            .into_iter()
-            .map(|(recipient, amount, lockup_date)| Allocation {
-                recipient,
-                amount: sol_to_lamports(amount),
-                lockup_date,
+        // We only support SOL token in "require lockup" mode.
+        rdr.deserialize()
+            .map(|recipient| {
+                let (recipient, amount, lockup_date): (String, f64, String) = recipient?;
+                match Pubkey::from_str(recipient.as_str()) {
+                    Ok(..) => (),
+                    Err(err) => return Err(Error::BadInputPubkeyError(err)),
+                }
+                if !lockup_date.is_empty() {
+                    match lockup_date.parse::<DateTime<Utc>>() {
+                        Ok(..) => (),
+                        Err(err) => return Err(Error::BadInputLockupDate(err)),
+                    }
+                } // empty lockup date means no lockup, it's okay to have only some lockups specified
```

Comment:
It seems`test_create_stake_allocations` test (the way it's currently written, with `require_lockup_heading=true` + the actual value present in csv being `""` for lockup date) implies this:
```rust
// empty lockup date means no lockup, it's okay to have only some lockups specified
```
yet, `require_lockup_heading` name kinda contradicts that,

I assume test case is correct, and we just want to change `require_lockup_heading` name into something like `with_lockup_date` ?

Reply:
I think `require_lockup_heading` is referring to the header row of the CSV only, although I do agree that it's not very clear. I'm open to changing this to `with_lockup` or `with_lockup_date`

Reply:
Renamed to `with_lockup` in https://github.com/solana-labs/solana/pull/33926/commits/30dfbaabae71d91c9da6907e82e465d22fee3fb1